yb-voyager/cmd/exportSchemaOptimizationReport.go

```diff
@@ -153,119 +165,109 @@ type SecondaryIndexToRangeChange struct {
+	return change
 }
 
 func buildShardingMviewRecommendationChange(shardedMviews []string, colocatedMviews []string) *ShardingRecommendationChange {
-	var appliedRecommendationMview *ShardingRecommendationChange
-	mviewFile := utils.GetObjectFilePath(filepath.Join(exportDir, "schema"), MVIEW)
+	referenceMviewFile := utils.GetObjectFilePath(filepath.Join(exportDir, "schema"), MVIEW)
 	//To mviews then add that change separately
-	if !utils.FileOrFolderExists(mviewFile) || len(shardedMviews) == 0 { // only display this in case there is any modifield sharded mview
+	if !utils.FileOrFolderExists(referenceMviewFile) || len(shardedMviews) == 0 { // only display this in case there is any modifield sharded mview
 		return nil
 	}
-	appliedRecommendationMview = NewAppliedShardingRecommendationChange(MVIEW)
 	// Get relative path from reports directory to the mview file
 	reportsDir := filepath.Join(exportDir, "reports")
-	relativePath, err := filepath.Rel(reportsDir, mviewFile)
-	if err != nil {
-		// Fallback to absolute path if relative path calculation fails
-		appliedRecommendationMview.ReferenceFile = mviewFile
-	} else {
-		appliedRecommendationMview.ReferenceFile = relativePath
+	relativePath, err := filepath.Rel(reportsDir, referenceMviewFile)
+	if err == nil {
+		//If no error, then use the relative path
+		referenceMviewFile = relativePath
 	}
-	appliedRecommendationMview.ReferenceFileDisplayName = filepath.Base(mviewFile)
-	appliedRecommendationMview.ShardedObjects = shardedMviews
-	appliedRecommendationMview.CollocatedObjects = colocatedMviews
-
-	return appliedRecommendationMview
+	change := NewAppliedShardingRecommendationChange(MVIEW, true)
+	change.ReferenceFile = referenceMviewFile
+	change.ReferenceFileDisplayName = "mview.sql"
+	change.ShardedObjects = shardedMviews
+	change.CollocatedObjects = colocatedMviews
+	return change
 }
 
 func buildSecondaryIndexToRangeChange(indexTransformer *sqltransformer.IndexFileTransformer) *SecondaryIndexToRangeChange {
 	if indexTransformer == nil {
 		return nil
 	}
 	if skipPerfOptimizations {
-		secondaryIndexToRangeChange := NewSecondaryIndexToRangeChange()
-		secondaryIndexToRangeChange.IsApplied = false
-		secondaryIndexToRangeChange.Description = "Range-sharded secondary indexes helps in better data distribution and performance. Due to the skip-performance-optimizations flag, all the btree indexes were not converted to range-sharded indexes. Modify the indexes to be range-sharded manually."
-		secondaryIndexToRangeChange.Title = "Secondary Indexes to be range-sharded - Not Applied"
-		return secondaryIndexToRangeChange
+		return NewSecondaryIndexToRangeChange(false, "", "", nil)
+	}
+	referenceIndexesFile := utils.GetObjectFilePath(filepath.Join(exportDir, "schema"), INDEX)
+	reportsDir := filepath.Join(exportDir, "reports")
+	relativePath, err := filepath.Rel(reportsDir, referenceIndexesFile)
+	if err == nil {
+		//If no error, then use the relative path
```

Comment:
I don't quite understand the need for the "relative path" . let's discuss this on call